### PR TITLE
Windows fixes

### PR DESCRIPTION
**Pull request by dirk-zimoch**
Fix for Issue #6.

### CODE REVIEW

File: caPutLogApp/caPutJsonLogTask.cpp [hunk 1]
```diff
@@ -11,7 +11,11 @@
  * - Initial version.
 */
 
+// Windows needs this macro to avoid confict with min/max macros
+#define NOMINMAX
```
Comment (mdavidsaver) [inline caPutLogApp/caPutJsonLogTask.cpp:15]: While defining `NOMINMAX` shouldn't do anything on targets other than Windows, I've opted for a conditional definition.  eg. in [`configure/CONFIG_SITE`](https://github.com/mdavidsaver/pvxs/blob/2ab0d662bf3f74f32915d0e07383d8c30217fdd5/configure/CONFIG_SITE#L45).

```makefile
USR_CPPFLAGS_WIN32 += -DNOMINMAX
```

File: caPutLogApp/caPutJsonLogTask.cpp [hunk 2]
```diff
@@ -11,7 +11,11 @@
...
+
 // Standard library imports
+#include <algorithm>  // Windows needs this for std::min/max
```
Comment (mdavidsaver) [inline caPutLogApp/caPutJsonLogTask.cpp:18]: This comment is a little misleading.  [`std::min()`](https://en.cppreference.com/w/cpp/algorithm/min) is defined in `<algorithm>` for all targets.

File: caPutLogApp/caPutJsonLogTask.cpp (hunk 1 above)
Comment (ralphlange) [reply caPutLogApp/caPutJsonLogTask.cpp:15]: +1

File: caPutLogApp/caPutJsonLogTask.cpp (hunk 2 above)
Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp:18]: But the other targets don't need `<algorithm>`.

Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp:18]: I can remove the comment. Just wanted to avoid that someone deletes it because "it is not needed".

Comment (mdavidsaver) [reply caPutLogApp/caPutJsonLogTask.cpp:18]: That this header may sometimes be implicitly included by others is incidental.  Standard C++ clearly defines the association between `std::min()` and `<algorithm>`.

imo. the best guard against regression would be to add CI tests.

Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp:18]: Feel free to add some. :-)

Comment (anjohnson) [reply caPutLogApp/caPutJsonLogTask.cpp:18]: Where is the code that requires `std::min()` when built for Windows?

There are macros for `min()` and `max()` defined in one of the source files inside a `#ifndef` test; maybe they should be replaced instead?

Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp]: The code lines are in `CaPutJsonLogTask::calculateMin` and `CaPutJsonLogTask::calculateMax` starting from line 594 to line 659.

File: caPutLogApp/caPutJsonLogTask.cpp (hunk 1 above)
Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp]: Done.

File: caPutLogApp/caPutJsonLogTask.cpp (hunk 2 above)
Comment (dirk-zimoch) [reply caPutLogApp/caPutJsonLogTask.cpp]: I have remove the comment.

Comment (mdavidsaver) [reply caPutLogApp/caPutJsonLogTask.cpp]: > Feel free to add some. :-)

I make a habit of adding test builds with [ci-scripts](https://github.com/epics-base/ci-scripts) when I have reason to touch a support module which doesn't have any.  I would encourage others to do the same.

**Review by anjohnson — APPROVED** [review]
These all look fine to me now.

**Review by ralphlange — APPROVED** [review]
Fine with me.

**Comment by ralphlange** [comment]
Merci!

**Pull request closed without merging**